src/components/PopupWithConfirmation.js

```diff
@@ -1,21 +1,20 @@
 import Popup from "./Popup.js";
 
 export default class PopupWithConfirmation extends Popup {
-  constructor({ popupSelector, handleConfirm }) {
+  constructor({ popupSelector }) {
     super(popupSelector);
-    this._handleConfirm = handleConfirm;
     this._confirmButton = this._popupElement.querySelector(".popup__button");
   }
 
   setSubmitAction(submitAction) {
```

Comment:
* There are errors when I delete 3 cards in a row http://joxi.ru/eAOOB91CGbnD6A 


You don't need to add the click handler every time. It makes the **memory leak**


```js

  setEventListeners() {
    super.setEventListeners();
//  here you set the click handler
```



And here will be only 


```js

  setSubmitAction(submitAction) {
    this._submitAction = submitAction
  }
```